dask/utils.py

```diff
@@ -198,17 +200,19 @@ def tmpdir(dir=None):
 
 @contextmanager
 def filetext(text, extension="", open=open, mode="w"):
-    with tmpfile(extension=extension) as filename:
-        f = open(filename, mode=mode)
-        try:
-            f.write(text)
-        finally:
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=FutureWarning)
+        with tmpfile(extension=extension) as filename:
```

Comment:
Instead of continuing to use `tmpfile` here and catching the `FutureWarning`, could we move away from `tmpfile` and use `TemporaryFile` or `NamedTemporaryFile` instead? This is preferred because, since `tmpfile` is deprecated, we'll be removing it in a a future `dask` release

Reply:
I wasn't able to get it working with `NamedTemporaryFile` as a context manager, but was able to use it by removing the file manually.